dataclients/kubernetes/definitions/routegroupvalidator.go

```diff
@@ -0,0 +1,204 @@
+package definitions
+
+import (
+	"errors"
+
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+)
+
+type RouteGroupValidator struct {
+	FilterRegistry filters.Registry
+}
+
+var defaultRouteGroupValidator = &RouteGroupValidator{}
+
+// validateRouteGroup validates a RouteGroupItem
+func ValidateRouteGroup(rg *RouteGroupItem) error {
+	return defaultRouteGroupValidator.Validate(rg)
+}
+
+func ValidateRouteGroupList(rgl RouteGroupList) error {
+	var err error
+	for _, rg := range rgl.Items {
+		err = errors.Join(err, defaultRouteGroupValidator.Validate(rg))
+	}
+	return err
+}
+
+func (rgv *RouteGroupValidator) Validate(item *RouteGroupItem) error {
+	err := rgv.basicValidation(item)
+	// if basic validation fails we don't need to run other validations, otherwise join errors.
+	if err != nil {
+		return err
+	}
+	return rgv.filtersValidation(item)
+}
+
+func (rgv *RouteGroupValidator) basicValidation(item *RouteGroupItem) error {
+	return item.validate()
+}
+
+func (rgv *RouteGroupValidator) filtersValidation(item *RouteGroupItem) error {
+	return validateRouteGroupFilters(item)
+}
```

Comment:
Lets inline these two., The code is small so there is no need to nest them. We can always refactor if either validation becomes more complex.